### PR TITLE
Use Go modules and fix linter warnings

### PR DESCRIPTION
**Pull request by howardburgess**
* Move from Dep to Go Modules and fix code for upgraded versions (Prometheus in particular)
* Fix various IntelliJ stylistic warnings throughout the code

I suggest to look at the commits individually.

Required by https://github.com/sky-uk/core-infrastructure/issues/3906

### CODE REVIEW

File: alb/alb.go [hunk 1]
```diff
@@ -25,15 +25,15 @@ func New(region string, targetGroupNames []string, targetGroupDeregistrationDela
 	}
 	initMetrics()
 	log.Infof("ALB frontend region: %s target groups: %v", region, targetGroupNames)
-	session, err := session.NewSession(&aws.Config{Region: &region})
+	awsSession, err := session.NewSession(&aws.Config{Region: &region})
```
Comment (howardburgess) [inline alb/alb.go]: IntelliJ: `Variable collides with imported package name`

File: alb/alb_test.go [hunk 2]
```diff
@@ -179,7 +175,7 @@ func TestErrorDescribingTargetGroups(t *testing.T) {
 	mockALB.mockDescribeTargetGroups([]string{"group"}, []string{"group-arn"}, nil, nil, errors.New("ba koom"))
 
 	//when
-	a.Start()
+	_ = a.Start()
```
Comment (howardburgess) [inline alb/alb_test.go]: IntelliJ: `Unhandled error`

File: controller/controller_test.go [hunk 3]
```diff
@@ -1254,7 +1254,7 @@ func createIngressesFixture(namespace string, host string, serviceName string, s
 	}
 
 	ingressDefinition := createIngressWithoutRules()
-	ingressRules := []v1beta1.IngressRule{}
+	var ingressRules []v1beta1.IngressRule
```
Comment (howardburgess) [inline controller/controller_test.go]: IntelliJ: `Empty slice declaration via literal`

File: dns/r53/client.go [hunk 4]
```diff
@@ -35,8 +35,9 @@ type client struct {
 // New creates a route53 client used to interact with aws
 func New(hostedZone string, retries int) Route53Client {
 	config := aws.Config{MaxRetries: aws.Int(retries)}
+	awsSession, _ := session.NewSession()
```
Comment (howardburgess) [inline dns/r53/client.go]: `session.New()` is deprecated

File: dns/r53/client_test.go [hunk 5]
```diff
@@ -71,11 +71,11 @@ func TestGetRecords(t *testing.T) {
 	// given
 	client, fake53 := createClient()
 	expectedRecords := []*route53.ResourceRecordSet{
-		&route53.ResourceRecordSet{
+		{
```
Comment (howardburgess) [inline dns/r53/client_test.go]: IntelliJ: `Redundant type`

File: elb/elb.go [hunk 6]
```diff
@@ -18,16 +18,16 @@ import (
 	"github.com/sky-uk/feed/util"
 )
 
-// ElbTag is the tag key used for identifying ELBs to attach to for a cluster.
-const ElbTag = "sky.uk/KubernetesClusterFrontend"
+// FrontendTag is the tag key used for identifying ELBs to attach to for a cluster.
+const FrontendTag = "sky.uk/KubernetesClusterFrontend"
```
Comment (howardburgess) [inline elb/elb.go]: IntelliJ: `Name starts with package name`

File: elb/status/status.go [hunk 7]
```diff
@@ -12,8 +12,8 @@ import (
 	"github.com/sky-uk/feed/controller"
 	"github.com/sky-uk/feed/elb"
 	"github.com/sky-uk/feed/k8s"
-	k8s_status "github.com/sky-uk/feed/k8s/status"
-	v1 "k8s.io/client-go/pkg/api/v1"
+	k8sStatus "github.com/sky-uk/feed/k8s/status"
```
Comment (howardburgess) [inline elb/status/status.go]: IntelliJ: `Eliminate snake case`

File: elb/status/status.go [hunk 8]
```diff
@@ -8,12 +8,12 @@ import (
 
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/aws/session"
-	aws_elb "github.com/aws/aws-sdk-go/service/elb"
+	awselb "github.com/aws/aws-sdk-go/service/elb"
```
Comment (howardburgess) [inline elb/status/status.go]: IntelliJ: `Eliminate snake case`

File: dns/dns_metrics.go [hunk 9]
```diff
@@ -13,42 +13,45 @@ var updateCount, failedCount, skippedCount prometheus.Counter
...
+		recordsGauge = prometheus.NewGauge(
 			prometheus.GaugeOpts{
 				Namespace:   metrics.PrometheusNamespace,
 				Subsystem:   metrics.PrometheusDNSSubsystem,
 				Name:        "route53_records",
 				Help:        "The current number of records.",
 				ConstLabels: metrics.ConstLabels(),
-			})).(prometheus.Gauge)
+			})
+		prometheus.MustRegister(recordsGauge)
```
Comment (mgreyo) [inline dns/dns_metrics.go]: There's quite a lot of repetition for this function call. Can we either use `prometheus.Register()` to register a single metric, or prepare a slice of `prometheus.Collector` to call `prometheus.MustRegister()` once with?

Comment (howardburgess) [reply dns/dns_metrics.go]: Most collectors use default options (`namespace`, `subsystem`, `...`) so I have introduced a helper function `prometheus.NewDefaultGauge()` to create them (see https://github.com/sky-uk/feed/pull/207/commits/42fb04086574d3c5aafef473692225ba3936b485)

**Comment by peterbale** [comment]
@howardburgess I'm going to be raising a relatively simple PR to fix https://github.com/sky-uk/core-infrastructure/issues/5172.  Can you wait for that to be merged too before creating a release after merging?

**Comment by howardburgess** [comment]
@peterbale Yes sure. In fact, I have another PR to raise following this, to enable support for NLBs. I was planning to do a release to include them both.

**Pull request merged**